app/electron/main.ts

```diff
@@ -415,6 +415,52 @@ function startElecron() {
       mainWindow = null;
     });
 
+    // Force Single Instance Application
+    const gotTheLock = app.requestSingleInstanceLock();
+    if (gotTheLock) {
+      app.on('second-instance', () => {
+        // Someone tried to run a second instance, we should focus our window.
+        if (mainWindow) {
+          if (mainWindow.isMinimized()) mainWindow.restore();
+          mainWindow.focus();
+        }
+      });
+    } else {
+      app.quit();
+      return;
+    }
+
+    /*
+    if a library is trying to open a url other than app url in electron take it 
+    to the default browser
+    */
+    mainWindow.webContents.on('will-navigate', (event, url) => {
+      if (url.startsWith(startUrl)) {
+        return;
+      }
+      shell.openExternal(url);
+    });
+
+    app.on('open-url', (event, url) => {
+      mainWindow?.focus();
+      let urlObj;
+      try {
+        urlObj = new URL(url);
+      } catch (e) {
+        throw new Error(`Invalid URL: ${url}`);
+      }
+      if (urlObj) {
```

Comment:
If there's an early return in the catch, maybe this is not needed. If Ts complains about it still, then you can check and early return before doing this new block.

Reply:
got it updated accordingly, thanks